sync fork with upstream

sunchayn opened the pull request.

dependabot[bot] committed:
Bump actions/checkout from 3 to 4

Bumps [actions/checkout](https://github.com/actions/checkout) from 3 to 4.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump stefanzweifel/git-auto-commit-action from 4 to 5

Bumps [stefanzweifel/git-auto-commit-action](https://github.com/stefanzweifel/git-auto-commit-action) from 4 to 5.
- [Release notes](https://github.com/stefanzweifel/git-auto-commit-action/releases)
- [Changelog](https://github.com/stefanzweifel/git-auto-commit-action/blob/master/CHANGELOG.md)
- [Commits](https://github.com/stefanzweifel/git-auto-commit-action/compare/v4...v5)

---
updated-dependencies:
- dependency-name: stefanzweifel/git-auto-commit-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

canvural committed:
cleanup

canvural committed:
feat: adjust to our needs - WIP

canvural committed:
feat: add filesystem driver

canvural committed:
fix: remove duplicated declaration

canvural committed:
fix: parentSpan is a property

canvural committed:
fix: default service and scope span name

canvural committed:
fix: start manual spand & register watchers

canvural committed:
feat: add manualStartAndEnd method

canvural committed:
fix: dont override attributes in manual spans

canvural committed:
feat: make Samplers configurable

canvural committed:
fix: return empty array when using createPayloadUsing hook

canvural committed:
fix: send payload as a raw body because it is already JSON encoded

PixellUp committed:
Fixed markdown instruction line style

freekmurze committed:
Merge pull request #31 from PixellUp/patch-1

Fixed markdown instruction line style

freekmurze committed:
Fix styling

dependabot[bot] committed:
Bump aglipanci/laravel-pint-action from 2.2.0 to 2.3.1

Bumps [aglipanci/laravel-pint-action](https://github.com/aglipanci/laravel-pint-action) from 2.2.0 to 2.3.1.
- [Release notes](https://github.com/aglipanci/laravel-pint-action/releases)
- [Commits](https://github.com/aglipanci/laravel-pint-action/compare/2.2.0...2.3.1)

---
updated-dependencies:
- dependency-name: aglipanci/laravel-pint-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #32 from spatie/dependabot/github_actions/aglipanci/laravel-pint-action-2.3.1

Bump aglipanci/laravel-pint-action from 2.2.0 to 2.3.1

SardorAlif committed:
hotfix/fix_basic_auth_error

freekmurze committed:
Merge pull request #35 from SardorAlif/hotfix/fix_basic_auth_error

hotfix/fix_basic_auth_error

freekmurze committed:
Fix styling

freekmurze committed:
Update CHANGELOG

Nielsvanpach committed:
Merge pull request #30 from spatie/dependabot/github_actions/stefanzweifel/git-auto-commit-action-5

Bump stefanzweifel/git-auto-commit-action from 4 to 5

Nielsvanpach committed:
Merge pull request #29 from spatie/dependabot/github_actions/actions/checkout-4

Bump actions/checkout from 3 to 4

laravel-shift committed:
Bump dependencies for Laravel 11

laravel-shift committed:
Update GitHub Actions for Laravel 11

freekmurze committed:
Merge pull request #37 from laravel-shift/l11-compatibility

Laravel 11.x Compatibility

dependabot[bot] committed:
Bump ramsey/composer-install from 2 to 3

Bumps [ramsey/composer-install](https://github.com/ramsey/composer-install) from 2 to 3.
- [Release notes](https://github.com/ramsey/composer-install/releases)
- [Commits](https://github.com/ramsey/composer-install/compare/v2...v3)

---
updated-dependencies:
- dependency-name: ramsey/composer-install
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump dependabot/fetch-metadata from 1.6.0 to 2.0.0

Bumps [dependabot/fetch-metadata](https://github.com/dependabot/fetch-metadata) from 1.6.0 to 2.0.0.
- [Release notes](https://github.com/dependabot/fetch-metadata/releases)
- [Commits](https://github.com/dependabot/fetch-metadata/compare/v1.6.0...v2.0.0)

---
updated-dependencies:
- dependency-name: dependabot/fetch-metadata
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

freekmurze committed:
Merge pull request #40 from spatie/dependabot/github_actions/dependabot/fetch-metadata-2.0.0

Bump dependabot/fetch-metadata from 1.6.0 to 2.0.0

freekmurze committed:
Merge pull request #38 from spatie/dependabot/github_actions/ramsey/composer-install-3

Bump ramsey/composer-install from 2 to 3

dependabot[bot] committed:
Bump aglipanci/laravel-pint-action from 2.3.1 to 2.4

Bumps [aglipanci/laravel-pint-action](https://github.com/aglipanci/laravel-pint-action) from 2.3.1 to 2.4.
- [Release notes](https://github.com/aglipanci/laravel-pint-action/releases)
- [Commits](https://github.com/aglipanci/laravel-pint-action/compare/2.3.1...2.4)

---
updated-dependencies:
- dependency-name: aglipanci/laravel-pint-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #41 from spatie/dependabot/github_actions/aglipanci/laravel-pint-action-2.4

Bump aglipanci/laravel-pint-action from 2.3.1 to 2.4

dependabot[bot] committed:
Bump dependabot/fetch-metadata from 2.0.0 to 2.1.0

Bumps [dependabot/fetch-metadata](https://github.com/dependabot/fetch-metadata) from 2.0.0 to 2.1.0.
- [Release notes](https://github.com/dependabot/fetch-metadata/releases)
- [Commits](https://github.com/dependabot/fetch-metadata/compare/v2.0.0...v2.1.0)

---
updated-dependencies:
- dependency-name: dependabot/fetch-metadata
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #43 from spatie/dependabot/github_actions/dependabot/fetch-metadata-2.1.0

Bump dependabot/fetch-metadata from 2.0.0 to 2.1.0

mmachatschek committed:
docs: Update requirements page to reflect the current system requirements

dependabot[bot] committed:
Bump dependabot/fetch-metadata from 2.1.0 to 2.2.0

Bumps [dependabot/fetch-metadata](https://github.com/dependabot/fetch-metadata) from 2.1.0 to 2.2.0.
- [Release notes](https://github.com/dependabot/fetch-metadata/releases)
- [Commits](https://github.com/dependabot/fetch-metadata/compare/v2.1.0...v2.2.0)

---
updated-dependencies:
- dependency-name: dependabot/fetch-metadata
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #47 from spatie/dependabot/github_actions/dependabot/fetch-metadata-2.2.0

Bump dependabot/fetch-metadata from 2.1.0 to 2.2.0

freekmurze committed:
Merge pull request #46 from mmachatschek/patch-1

docs: Update requirements page to reflect the current system requirements

freekmurze committed:
Fix styling

freekmurze committed:
Update CHANGELOG

dependabot[bot] committed:
Bump dependabot/fetch-metadata from 2.2.0 to 2.3.0

Bumps [dependabot/fetch-metadata](https://github.com/dependabot/fetch-metadata) from 2.2.0 to 2.3.0.
- [Release notes](https://github.com/dependabot/fetch-metadata/releases)
- [Commits](https://github.com/dependabot/fetch-metadata/compare/v2.2.0...v2.3.0)

---
updated-dependencies:
- dependency-name: dependabot/fetch-metadata
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #50 from spatie/dependabot/github_actions/dependabot/fetch-metadata-2.3.0

Bump dependabot/fetch-metadata from 2.2.0 to 2.3.0

dependabot[bot] committed:
Bump aglipanci/laravel-pint-action from 2.4 to 2.5

Bumps [aglipanci/laravel-pint-action](https://github.com/aglipanci/laravel-pint-action) from 2.4 to 2.5.
- [Release notes](https://github.com/aglipanci/laravel-pint-action/releases)
- [Commits](https://github.com/aglipanci/laravel-pint-action/compare/2.4...2.5)

---
updated-dependencies:
- dependency-name: aglipanci/laravel-pint-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

freekmurze committed:
Merge pull request #51 from spatie/dependabot/github_actions/aglipanci/laravel-pint-action-2.5

Bump aglipanci/laravel-pint-action from 2.4 to 2.5

freekmurze committed:
Fix styling

laravel-shift committed:
Bump dependencies for Laravel 12

laravel-shift committed:
Update GitHub Actions for Laravel 12

freekmurze committed:
Update run-tests.yml

freekmurze committed:
Update run-tests.yml

freekmurze committed:
Update composer.json

freekmurze committed:
Merge pull request #52 from laravel-shift/l12-compatibility

Laravel 12.x Compatibility

freekmurze committed:
Update CHANGELOG

jimirobaer committed:
Update README.md

jimirobaer committed:
Update README.md

dependabot[bot] committed:
Bump dependabot/fetch-metadata from 2.3.0 to 2.4.0

Bumps [dependabot/fetch-metadata](https://github.com/dependabot/fetch-metadata) from 2.3.0 to 2.4.0.
- [Release notes](https://github.com/dependabot/fetch-metadata/releases)
- [Commits](https://github.com/dependabot/fetch-metadata/compare/v2.3.0...v2.4.0)

---
updated-dependencies:
- dependency-name: dependabot/fetch-metadata
  dependency-version: 2.4.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #54 from spatie/dependabot/github_actions/dependabot/fetch-metadata-2.4.0

Bump dependabot/fetch-metadata from 2.3.0 to 2.4.0

sunchayn committed:
Fix styling

sunchayn committed:
Merge branch 'main' into feature/sync-fork-with-upstream

sunchayn committed:
Fix styling

canvural commented:
I don't think we need to care about tests too much. Probably they were not running previously too 😅

sunchayn committed:
test: fix some tests

sunchayn committed:
fix: resolve class using the container

the $this in the macro will be a `PendingRequest` instance

sunchayn committed:
Fix styling

sunchayn commented:
> I don't think we need to care about tests too much. Probably they were not running previously too 😅

haha yes it seems that they will always be failing due to signature changes. However, I fixed a few and removed ones that are testing things we no longer have.

however, I think this was an actual bug [fix: resolve class using the container](https://github.com/StuDocu/laravel-open-telemetry/pull/1/commits/70aaf77c07e51df9f5f6bbbd4d33ee0df1145d42)

sunchayn commented:
@canvural feel free to merge and release if all is good 👌
